apps/finance/app/src/components/Transfers.js

```diff
@@ -29,236 +30,220 @@ const TRANSFER_TYPES = [
     start: null,
     end: null,
-  },
-  selectedToken: 0,
-  selectedTransferType: 0,
-  displayedTransfers: TRANSFERS_PER_PAGE,
-}
-
-class Transfers extends React.PureComponent {
-  state = {
-    ...initialState,
-    filtersOpened: !this.props.compactMode,
-  }
-
-  handleToggleFiltersClick = () => {
-    this.setState(({ filtersOpened }) => ({ filtersOpened: !filtersOpened }))
-  }
-  handleTokenChange = index => {
-    this.setState({
-      selectedToken: index,
-      displayedTransfers: TRANSFERS_PER_PAGE,
-    })
-  }
-  handleTransferTypeChange = index => {
-    this.setState({
-      selectedTransferType: index,
-      displayedTransfers: TRANSFERS_PER_PAGE,
-    })
-  }
-  handleResetFilters = () => {
-    this.setState(({ selectedDateRange }) => ({
-      ...initialState,
-      selectedDateRange,
-    }))
-  }
-  handleDateRangeChange = selectedDateRange => {
-    this.setState({ selectedDateRange })
-  }
-  encodeDataToCsv = (data, tokenDetails) => {
-    const csvContent = [
-      'data:text/csv;charset=utf-8,Date,Source/Recipient,Reference,Amount',
-    ]
-      .concat(
-        data.map(
-          ({
-            date,
-            numData: { amount },
-            reference,
-            isIncoming,
-            entity,
-            token,
-          }) => {
-            const { symbol, decimals } = tokenDetails[toChecksumAddress(token)]
-            const formattedAmount = formatTokenAmount(
-              amount,
-              isIncoming,
-              decimals,
-              true,
-              { rounding: 5 }
-            )
-            return `${formatDate(
-              date
-            )},${entity},${reference},${`${formattedAmount} ${symbol}`}`
-          }
-        )
-      )
-      .join('\n')
-    return window.encodeURI(csvContent)
-  }
-  getCsvFilename = () => {
-    const { selectedDateRange } = this.state
-    const start = format(selectedDateRange.start, 'yyyy-MM-dd')
-    const end = format(selectedDateRange.end, 'yyyy-MM-dd')
-    return `transfers_${start}_to_${end}.csv`
-  }
-  showMoreTransfers = () => {
-    this.setState(prevState => ({
-      displayedTransfers: prevState.displayedTransfers + TRANSFERS_PER_PAGE,
-    }))
-  }
+  })
 
-  // Filter transfer based on the selected filters
-  getFilteredTransfers({
+  const handleToggleFiltersClick = React.useCallback(() => {
+    setFiltersOpened(!filtersOpened)
+  })
+  const handleTokenChange = React.useCallback(index => {
+    setSelectedToken(index)
+    setDisplayedTransfers(TRANSFERS_PER_PAGE)
+  })
+  const handleTransferTypeChange = React.useCallback(index => {
+    setSelectedTransferType(index)
+    setDisplayedTransfers(TRANSFERS_PER_PAGE)
+  })
+  const handleResetFilters = React.useCallback(() => {
+    setDisplayedTransfers(TRANSFERS_PER_PAGE)
+    setSelectedToken(0)
+    setSelectedTransferType(0)
+  })
+  const getCsvFilename = React.useCallback(() => {
+    let filename = 'transfers.csv'
```

Comment:
What do you think about adding the current export date to the default filename?

Reply:
This is the filename when a date range has not been selected.